tests/periph_i2c/tests/test.py

```diff
@@ -0,0 +1,328 @@
+    t.notes += 'Cannot test since it can still be used without mutex\n'
+    t.notes += 'Cannot test expected timeout due to no reset\n'
+    t.notes += 'Do we need double release failure check?\n'
+    t.notes += 'Does not check multiple busses acquired\n'
+
+    setup_test(t)
+
+    num_of_dev = t.run_test(i2c.get_devs(), "Success")[0]
+
+    for d_num in range(0, num_of_dev):
+        t.run_test(i2c.release(d_num), "Success")
+
+        t.run_test(i2c.acquire(d_num), "Success")
+
+        if (d_num == 0):
+            t.run_test(i2c.read_reg(d_num, __BPT_ADDR, __BPT_USER_REG),
+                       "Success", [0])
+
+        t.run_test(i2c.release(d_num), "Success")
+
+        t.skip_test("i2c.read_byte(d_num, __BPT_ADDR), Error")
+        t.skip_test("i2c.acquire(d_num), Success")
+        t.skip_test("i2c.acquire(d_num), Timeout")
+
+    return t
+
+
+def read_test():
+    cmd_log = list()
+    t = Test('read test', 'Tests the read functionality for default i2c bus \
+(0) with the bpt', cmd_log)
+    t.notes += 'Does not check flags\n'
+
+    setup_test(t)
+
+    d_num = 0
+    t.run_test(i2c.release(d_num), "Success")
+    t.run_test(i2c.acquire(d_num), "Success")
+    t.run_test(i2c.read_reg(d_num, __BPT_ADDR, __BPT_USER_REG), "Success", [0])
+    t.run_test(i2c.read_reg(d_num, __BPT_ADDR+1, __BPT_USER_REG), "Error")
+    t.run_test(i2c.read_reg(d_num, __BPT_ADDR, __BPT_USER_REG+1),
+               "Success", [1])
+    t.run_test(i2c.read_regs(d_num, __BPT_ADDR, __BPT_USER_REG, 1),
+               "Success", [0])
+    t.run_test(i2c.read_regs(d_num, __BPT_ADDR, __BPT_USER_REG+1, 1),
+               "Success", [1])
+    t.run_test(i2c.read_regs(d_num, __BPT_ADDR, __BPT_USER_REG, 2),
+               "Success", [0, 1])
+    t.run_test(i2c.read_regs(d_num, __BPT_ADDR, __BPT_USER_REG, 3),
+               "Success", [0, 1, 2])
+    t.run_test(i2c.read_regs(d_num, __BPT_ADDR-1, __BPT_USER_REG, 1), "Error")
+    t.run_test(i2c.read_byte(d_num, __BPT_ADDR,), "Success", [0])
+    t.run_test(i2c.read_byte(d_num, __BPT_ADDR-2), "Error")
+    t.run_test(i2c.read_bytes(d_num, __BPT_ADDR, 1), "Success", [0])
+    t.run_test(i2c.read_bytes(d_num, __BPT_ADDR, 2), "Success", [0, 1])
+    t.run_test(i2c.read_bytes(d_num, __BPT_ADDR, 3), "Success", [0, 1, 2])
+    t.run_test(i2c.read_bytes(d_num, __BPT_ADDR+4, 3), "Error")
+    t.run_test(i2c.release(d_num), "Success")
+    return t
+
+
+def check_write_reg(t, d_num, addr, reg, val):
+    t.run_test(i2c.write_reg(d_num, addr, reg, val[0]), "Success")
+    t.run_test(i2c.read_reg(d_num, addr, reg), "Success", val)
+
+
+def check_write_regs(t, d_num, addr, reg, val):
+    t.run_test(i2c.write_regs(d_num, addr, reg, val), "Success")
+    t.run_test(i2c.read_regs(d_num, addr, reg, len(val)), "Success", val)
+
+
+def write_test():
+    cmd_log = list()
+    t = Test('write test', 'Tests the write functionality for default i2c bus \
+(0) with the bpt', cmd_log)
```

Comment:
indent please

Reply:
Then the output includes the indents I think...